### PR TITLE
more socket stuff

### PR DESCRIPTION
**Pull request by pdlla**

### CODE REVIEW

File: src/p2p/potato_socket.erl [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    {ok, LSock} = gen_tcp:listen(5678, [binary, {active, true}]),
+    tcp_listener(LSock).
+
+tcp_listener(LSock) ->
+    %io:fwrite("listening\n"),
+    % for some reason this will return {error, closed} when handle_server_side returns...
+    {ok, Sock} = gen_tcp:accept(LSock),
+    io:fwrite("accept\n"),
+    spawn(fun() -> tcp_listener(LSock) end),
+    handle_server_side(Sock).
```
Comment (antonbobkov) [inline src/p2p/potato_socket.erl]: Should you spawn server side and tail-recurse tcp_listener?

File: src/p2p/potato_socket.erl [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    end.
+
+start_server() ->
+    io:fwrite("starting server \n"),
+    {ok, LSock} = gen_tcp:listen(5678, [binary, {active, true}]),
+    tcp_listener(LSock).
+
+tcp_listener(LSock) ->
+    %io:fwrite("listening\n"),
+    % for some reason this will return {error, closed} when handle_server_side returns...
```
Comment (antonbobkov) [inline src/p2p/potato_socket.erl]: mm, not sure why

File: src/p2p/potato_socket.erl [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    {ok, Sock} = gen_tcp:connect("localhost", 5678, [binary, {active, true}]),
+    gen_tcp:send(Sock, "Some Data"),
+    handle_client_side(Sock).
+
+handle_client_side(Socket) ->
+    %io:fwrite("handling socket client\n"),
+    receive
+        {tcp, Socket, Msg} ->
+            %%gen_tcp:send(Socket, Msg),
+            io:fwrite("client got message ~p\n", [Msg]),
```
Comment (antonbobkov) [inline src/p2p/potato_socket.erl]: you can also do ~n

File: src/p2p/potato_socket.erl [hunk 4]
```diff
@@ -0,0 +1,56 @@
+-module(potato_socket).
+
+-export([start/0, start_server/0, start_client/0, handle_server_side/1, handle_client_side/1]).
+
+-include_lib("stdlib/include/assert.hrl").
+
+start() ->
+    io:fwrite("hello, world\n"),
+    spawn(?MODULE, start_server, []),
+    timer:sleep(100), % need to sleep otherwise there is a race cond between server/client
```
Comment (antonbobkov) [inline src/p2p/potato_socket.erl]: you could wait for a message *started* from the server

**Comment by antonbobkov** [comment]
MUCH PRODUCTIVE

File: src/p2p/potato_socket.erl (hunk 4 above)
Comment (pdlla) [reply src/p2p/potato_socket.erl]: D:

**Pull request merged**